modelforge/potential/models.py

```diff
@@ -686,124 +682,180 @@ def _input_checks(self, data: Union[NNPInput, NamedTuple]):
             The dataset statistics.
         """
         super().__init__()
 
+        self._registered_properties: List[str] = []
+
+        # operations that use nn.Sequence to pass the output of the model to the next
+        self.registered_chained_operations = ModuleDict()
+
+        self.dataset_statistic = dataset_statistic
+
         self._initialize_postprocessing(
-            processing_operation,
-            readout_operation,
-            dataset_statistic,
+            postprocessing_parameter,
         )
 
+    def _get_mean_and_stddev_of_dataset(self) -> Tuple[float, float]:
+
+        if self.dataset_statistic is None:
+            mean = 0.0
+            stddev = 1.0
+            log.warning(
+                f"No mean and stddev provided for dataset. Setting to default value {mean=} and {stddev=}!"
+            )
+        else:
+            training_dataset_statistics = self.dataset_statistic[
+                "training_dataset_statistics"
+            ]
+            mean = unit.Quantity(
+                training_dataset_statistics["per_atom_energy_mean"]
+            ).m_as(unit.kilojoule_per_mole)
+            stddev = unit.Quantity(
+                training_dataset_statistics["per_atom_energy_stddev"]
+            ).m_as(unit.kilojoule_per_mole)
+        return mean, stddev
+
     def _initialize_postprocessing(
         self,
-        processing_operation: List[Dict[str, str]],
-        readout_operation: List[Dict[str, str]],
-        dataset_statistic,
+        postprocessing_parameter: Dict[str, Dict[str, bool]],
     ):
         from .processing import (
             FromAtomToMoleculeReduction,
             ScaleValues,
             CalculateAtomicSelfEnergy,
         )
 
-        # initialize per atom processing
-        work_to_be_done_per_property = []
-        props = []
-        for proc in processing_operation:
-            if proc["step"] == "normalization":
-                if dataset_statistic is None:
-                    log.warning(
-                        f"No mean and stddev provided for property {proc['in']}. Setting to default values!"
-                    )
-                    mean = 0.0
-                    stddev = 1.0
-                else:
-                    atomic_energies_stats = dataset_statistic["atomic_energies_stats"]
-                    mean = unit.Quantity(atomic_energies_stats[proc["mean"]]).m_as(
-                        unit.kilojoule_per_mole
+        # register properties
+        for property in postprocessing_parameter:
+            if property.lower() in self._SUPPORTED_PROPERTIES:
+                self._registered_properties.append(property.lower())
+            else:
+                raise ValueError(
+                    f"Property {property} is not supported. Supported properties are {self._SUPPORTED_PROPERTIES}"
+                )
+
+        # register operations
+        for property, operations in postprocessing_parameter.items():
```

Comment:
since this is looping over what was defined in the toml file, not over `self._SUPPORTED_PROPERTIES` (and see if the property key exists in `self._SUPPORTED_PROPERTIES`), doesn't this mean the order things are defined will influence the order of operations?  I thought we had decided we wanted it to be order independent (where the order of operations is defined within the code). 

 Right now we have't necessarily defined any sort of chained operations (everything is independent) so I don't think it matters yet, but I think it would be good to change this loop around future proof things or alternatively change the loops starting on line 749 (where we loop over the input operations) or 847 (where we finally call them).

Reply:
Thanks for noting this! I was planning to loop through `self._SUPPORTED_PROPERTIES` but didn't change this it seems. I will modify the code. In that way the order can't be modified at runtime.

Reply:
We are now looping over `self._SUPPORTED_PROPERTIES`! But I think, in general, the best way to mitigate this is to define a sequence of operations per property that has an order but doesn't interact with the next sequence of operations for the same or different property. This is something that we have to ensure in the module and can't be user defined.